### PR TITLE
An attempt to fix some theming issues and to make JS UI logic more consistent

### PR DESCRIPTION
**Pull request by aparatext**
Hi. I've attempted to fix some dark theme issues and to make logic JS more consistent, but I might have created some regressions. I couldn't get the identification to work, so I couldn't test the plant page UI. I'd appreciate any feedback. Thanks for building this useful app!

### CODE REVIEW

**Comment by patrickjane** [comment]
Hey, thanks for your efforts. I have tested the app in dark mode before submitting, and for me it looked good, although I would not say that something couldn't have slipped through. What issues did you have exactly? Did you use UTT for setting dark mode? Did you use a specific theme?

I find it a bit hard to follow the several aspects of your changes, since everything is done within one commit, and you're mixing theme changes, bug fixes? and style changes. And, to be honest, the PR feels a bit offensive, since instead of a comment like "Maybe you could think about doing xxxx" you submitted an overhaul of the (working) code. While I appreciate your efforts, I won't merge a PR with purely cosmetic/non-functional changes, since style is probably subject to personal taste and preferences, especially when there is no widely adopted/agreed style guide. While I fully embrace the usefulness of uniform code styles, and I follow it strictly in bigger projects, or at my projects at work (3000+ company), I handle it somewhat more relaxed when we talk a but a really small app, with really small functionality, which is only maintained by a single person, which I made in my free time. Of course one could argue that it would be best to *always* follow strict style guides - and while I actually consider doing so in the C++ parts - but as I mentioned, I think there is limited usefulness in QML.
Since the app was made in only about 5-7 days, I admit that I forgot some optional ; here and there, thanks for pointing out, gonna adopt your changes here.

I also saw that you left some comments in code here and there, and there were spots where you did not seem to understand the app logic. It seems like you wanted to fix something, but I cannot see from the review what was fixed, or *if* it was fixed.

As a compromise:
1) Let me know about the actual theme issues, and I will be happy to adopt your changes
2) I'm gonna run QtCreator auto-formatting for the next release, which I haven't, since I don't work with QtCreator, so the style should be more uniform
3) If you found potential bugs in the code, please clearly point out what functionality is not working, and I will provide a fix, or adopt a dedicated fix from your side.

**Comment by aparatext** [comment]
I'm sorry for putting everything in one commit. I just tried fixing some minor UI nitpicks and changed some logic along the way. I'm sorry if this whole PR looks pretentious and I've only introduced bugs with those style changes. I didn't quite understand some logic where I left the `// FIXME` comments.

The only minor thing I found is that some things don't follow the theme palette, leading to weird problems with the dark theme. Everything else is really just personal preference. I thought buttons could use `theme.palette.normal.positive`.

This PR can just be closed since most changes in it were useless. I think I should close this PR and open a new one, solving only these theme issues.

I really don't want to sound like I know better. Sorry for imposing my preference on others.

**Comment by patrickjane** [comment]
Alright I see.
BTW is the `Theme` type actually available with both lowercase and uppercase names? I am using `Theme.palette.normal.baseText`, while you had `theme.palette.normal.positive`.

**Comment by patrickjane** [comment]
@walking-octopus Okay so I have added code reformatting for all source files, C++ and QML. For QML, unfortunately, I was not able to find a formatter for VSCode which works well, so I ran QtCreator and reformatted the QML using QtCreator built in functionality.

**Comment by aparatext** [comment]
The icons have poor contrast. No one answered the question on Telegram, but I've found an example in Custom Phablet Tools.
[Example](https://framagit.org/ubuntouch-fr-dev/custom-phablet-tools/-/blob/master/qml/IconsView.qml#L90)

**Comment by patrickjane** [comment]
Yeah it's really weird, I don't understand why the icon color is barely visible with default settings. I also asked this on telegram in the past, and no one knew why.
To be honest, it feels like either a pretty severe bug, or something obvious that I am missing.

Anyway, I have made a custom button which looks like the original one, except that the icon color is clearly visible. I took a look at your example, but the code piece you sent just makes a pretty round button, so this seemed pretty off from what I wanted to do.
